pkg/alertmanager/multitenant.go

```diff
@@ -431,6 +439,8 @@ func (am *MultitenantAlertmanager) deleteUser(userID string) {
 }
 
 func (am *MultitenantAlertmanager) newAlertmanager(userID string, amConfig *amconfig.Config) (*Alertmanager, error) {
+	reg := prometheus.NewRegistry()
+	am.metrics.addUserRegistry(userID, reg)
```

Comment:
This is tricky actually... by removing the registry, you can break some metrics. It's ok if per-user counter disappears, and then reappears from 0. But it's not OK for global counters (`alertmanager_alerts_received_total`, `alertmanager_alerts_invalid_total`), summaries (`alertmanager_nflog_gc_duration_seconds`) or histograms (`alertmanager_notification_latency_seconds`) to go down.

For global counters like that, you need to preserve related values from the registry to some local counters in `alertmanagerMetrics` and then add those stored values when sending metrics in `Collect` method. metrics_helper.go has `SummaryData` and `HistogramData` that can help with that.

Reply:
If user will not be re-added later, it might actually be easier just to leave it in the map.

However, if same user can reappear later, that will not work, as previous registry cannot be passed to new AlertManager – registering metrics would panic :(

Reply:
In that case the best course of action would be to deactivate the users Alertmanager instead of delete it. It is totally conceivable for a user to delete then recreate an Alertmanager. It is much easier to reactivate an Alertmanager with a new config instead of going through the rigamarole of handling this in the registry.

Reply:
I'm going to wait for #2125 to be merged before handling solving this issue.

Reply:
@jtlisi #2125 is merged done. I think we can now address this issue (and BTW, I do agree that de-activating/re-activating looks a reasonable solution *if* doesn't have side effects).

Reply:
I have updated this PR to only pause a users alertmanager and restart it when a new config is applied.